sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -1298,13 +1303,15 @@ export class BlobClient extends StorageClient {
       if (e.statusCode === 404) {
         // Expected exception when checking blob existence
         return false;
-      } else if (
-        e.statusCode === 409 &&
-        (e.details.errorCode === BlobUsesCustomerSpecifiedEncryptionMsg ||
-          e.details.errorCode === BlobDoesNotUseCustomerSpecifiedEncryption)
-      ) {
-        // Expected exception when checking blob existence
-        return true;
+      } else if (e.statusCode === 409) {
+        const errorCode = e.details?.errorCode ?? e.details?.code;
```

Comment:
suggestion: add a small utility helper like `getErrorCode` to encapsulate this logic in case we have to extend it again in the future